### ISSUE TITLE
Missing version bound for `void`

### ISSUE THREAD

**Issue by hvr** [Atry/dsl#2]
The latest release, `do-notation-dsl-0.1.0.2` added a dependency on `void` but unfortunately didn't specify any versions bounds (we expect the PVP to be followed on Hackage and this implies proper use of version bounds on dependencies, see also https://pvp.haskell.org/faq/).

In particular, by leaving `void` unconstrained you're stating that *any* version of `void` will do; which is provably false:

```
Build profile: -w ghc-7.8.4 -O1
In order, the following will be built (use -v for more details):
 - do-notation-dsl-0.1.0.2 (lib) (first run)
Configuring library for do-notation-dsl-0.1.0.2..
Preprocessing library for do-notation-dsl-0.1.0.2..
Building library for do-notation-dsl-0.1.0.2..
[ 1 of 10] Compiling Control.Dsl.Cont ( src/Control/Dsl/Cont.hs, /tmp/do-notation-dsl-0.1.0.2/dist-newstyle/build/x86_64-linux/ghc-7.8.4/do-notation-dsl-0.1.0.2/build/Control/Dsl/Cont.o )
[ 2 of 10] Compiling Control.Dsl.Dsl  ( src/Control/Dsl/Dsl.hs, /tmp/do-notation-dsl-0.1.0.2/dist-newstyle/build/x86_64-linux/ghc-7.8.4/do-notation-dsl-0.1.0.2/build/Control/Dsl/Dsl.o )

src/Control/Dsl/Dsl.hs:21:10: Warning: Unrecognised pragma
[ 3 of 10] Compiling Control.Dsl.Return ( src/Control/Dsl/Return.hs, /tmp/do-notation-dsl-0.1.0.2/dist-newstyle/build/x86_64-linux/ghc-7.8.4/do-notation-dsl-0.1.0.2/build/Control/Dsl/Return.o )

src/Control/Dsl/Return.hs:24:32: Not in scope: ‘absurd’
```

long story short, please set something like

```
build-depends: void >= 0.7.2 && < 0.8
```

in your .cabal file

This time I'd like to give you the opportunity to perform the revision yourself, just go to

https://hackage.haskell.org/package/do-notation-dsl-0.1.0.2/do-notation-dsl.cabal/edit

and add the version constraint after `void`

Please let me know if you have any questions!

**Comment by Atry** [comment]
You are very kind. I wonder how to verify if the dependency is correct. It seems that Haskell Matrix CI does not test with old dependency versions.

**Comment by Atry** [comment]
I removed the dependency to `void` (cff7cf1078c019bfa0b390aebebac0063eccac2d), instead, `base >= 4.8` is required (8ad70d9e6fd3ec46f4082977bccf4cdabca77ecb).

[do-notation-dsl-0.1.0.2](https://hackage.haskell.org/package/do-notation-dsl-0.1.0.2) is deprecated since the dependency to `void` breaks forward compatibility to 0.1.0.1 therefore it does not conform PVP.
